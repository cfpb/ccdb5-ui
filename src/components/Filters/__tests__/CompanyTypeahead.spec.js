import {
  CompanyTypeahead,
  mapDispatchToProps,
  mapStateToProps,
} from '../CompanyTypeahead';
import React from 'react';
import { shallow } from 'enzyme';

/**
 *
 */
function setupEnzyme() {
  const props = {
    queryString: '?foo=bar&baz=qaz',
    typeaheadSelect: jest.fn(),
  };

  const target = shallow(<CompanyTypeahead {...props} />);

  return {
    props,
    target,
  };
}

describe('component::CompanyTypeahead', () => {
  describe('Typeahead interface', () => {
    beforeEach(() => {
      global.fetch = jest.fn().mockImplementation((url) => {
        expect(url).toContain('@@API_suggest_company/?foo=bar&baz=qaz&text=');

        return new Promise((resolve) => {
          resolve({
            json: function () {
              return ['foo', 'bar', 'baz', 'qaz'];
            },
          });
        });
      });
    });

    describe('_onInputChange', () => {
      it('provides a promise', () => {
        const { target } = setupEnzyme();
        const actual = target.instance()._onInputChange('mo');
        expect(actual.then).toBeInstanceOf(Function);
      });
    });

    describe('_renderOption', () => {
      it('produces a custom component', () => {
        const { target } = setupEnzyme();
        const option = {
          key: 'Foo',
          label: 'foo',
          position: 0,
          value: 'FOO',
        };
        const { value, component } = target.instance()._renderOption(option);
        expect(value).toEqual('Foo');
        expect(component).toMatchObject({
          props: {
            label: 'foo',
            position: 0,
            value: 'FOO',
          },
        });
      });
    });

    describe('_onOptionSelected', () => {
      it('checks the filter associated with the option', () => {
        const { target, props } = setupEnzyme();
        target.instance()._onOptionSelected({ key: 'foo' });
        expect(props.typeaheadSelect).toHaveBeenCalledWith('foo');
      });
    });
  });

  describe('mapDispatchToProps', () => {
    it('hooks into addMultipleFilters', () => {
      const dispatch = jest.fn();
      mapDispatchToProps(dispatch).typeaheadSelect('foo');
      expect(dispatch.mock.calls.length).toEqual(1);
    });
  });

  describe('mapStateToProps', () => {
    it('maps state and props', () => {
      const state = {
        aggs: {
          zip_code: [123, 456, 789],
        },
        query: {
          focus: false,
          lens: '',
          queryString: '?dsfds=2232',
          searchAfter: '12344_1233',
          state: ['TX', 'FL'],
          zip_code: '',
        },
      };
      const actual = mapStateToProps(state);
      expect(actual).toEqual({
        isDisabled: false,
        queryString: '?state=TX&state=FL',
      });
    });
  });
});
