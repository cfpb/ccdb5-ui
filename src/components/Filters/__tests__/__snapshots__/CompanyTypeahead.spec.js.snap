// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component::CompanyTypeahead snapshots renders disabled without crashing 1`] = `
<section
  className="typeahead "
  onBlur={[Function]}
  onFocus={[Function]}
>
  <div
    className="m-btn-inside-input input-contains-label"
  >
    <div
      className="input-contains-label_before input-contains-label_before__search"
    >
      <svg
        className="cf-icon-svg"
        viewBox="0 0 880.6 1200"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M860.1 879.2L645.7 664.8c90.8-136.8 76-323-44.6-443.5-137.5-137.5-360.5-137.5-498 0s-137.5 360.5 0 498c118.5 118.4 303.9 137 443.5 44.6L761 978.3c27.3 27.3 71.7 27.3 99 0s27.4-71.8.1-99.1zm-508-116.9C191.4 762 61.3 631.5 61.6 470.8c.3-160.7 130.8-290.8 291.5-290.5s290.8 130.8 290.5 291.5c-.2 118.2-71.9 224.6-181.5 269.1-34.9 14.2-72.3 21.5-110 21.4z"
        />
      </svg>
    </div>
    <label
      className="u-visually-hidden"
      htmlFor="company-typeahead-foobar"
    >
      Start typing to begin listing companies
    </label>
    <input
      autoComplete="off"
      className="a-text-input"
      disabled={true}
      id="company-typeahead-foobar"
      onChange={[Function]}
      onKeyDown={[Function]}
      placeholder="Enter company name"
      type="text"
      value=""
    />
  </div>
</section>
`;

exports[`component::CompanyTypeahead snapshots renders without crashing 1`] = `
<section
  className="typeahead "
  onBlur={[Function]}
  onFocus={[Function]}
>
  <div
    className="m-btn-inside-input input-contains-label"
  >
    <div
      className="input-contains-label_before input-contains-label_before__search"
    >
      <svg
        className="cf-icon-svg"
        viewBox="0 0 880.6 1200"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M860.1 879.2L645.7 664.8c90.8-136.8 76-323-44.6-443.5-137.5-137.5-360.5-137.5-498 0s-137.5 360.5 0 498c118.5 118.4 303.9 137 443.5 44.6L761 978.3c27.3 27.3 71.7 27.3 99 0s27.4-71.8.1-99.1zm-508-116.9C191.4 762 61.3 631.5 61.6 470.8c.3-160.7 130.8-290.8 291.5-290.5s290.8 130.8 290.5 291.5c-.2 118.2-71.9 224.6-181.5 269.1-34.9 14.2-72.3 21.5-110 21.4z"
        />
      </svg>
    </div>
    <label
      className="u-visually-hidden"
      htmlFor="company-typeahead-foobar"
    >
      Start typing to begin listing companies
    </label>
    <input
      autoComplete="off"
      className="a-text-input"
      id="company-typeahead-foobar"
      onChange={[Function]}
      onKeyDown={[Function]}
      placeholder="Enter company name"
      type="text"
      value=""
    />
  </div>
</section>
`;
